futures-core/src/task_impl/mod.rs

```diff
@@ -266,30 +199,16 @@ impl<T: ?Sized> Spawn<T> {
         self.enter(BorrowedUnpark::new(&mk, id), f)
     }
 
-    /// TODO: dox
-    pub fn with_task_data<F, R>(&mut self, f: F) -> R
```

Comment:
The RFC doesn't actually state what the changes to all of the API's should be, but this one doesn't seem useful anymore, given that the context explicitly contains task data.

Reply:
👍